layout-service/src/main/java/io/meeds/layout/service/PageLayoutService.java

```diff
@@ -291,8 +291,13 @@ public void updatePageApplicationPreferences(PageKey pageKey,
                                                       applicationId,
                                                       pageKey));
     }
-    Portlet portletPreferences = new Portlet();
-    preferences.forEach(preference -> portletPreferences.setValue(preference.getName(), preference.getValue()));
+    Portlet portletPreferences = layoutService.load(application.getState());
+    if (portletPreferences == null) {
+      portletPreferences = new Portlet();
+    }
+    for (PortletInstancePreference preference : preferences) {
+      portletPreferences.setValue(preference.getName(), preference.getValue());
+    }
```

Comment:
Please keep it as is, since it must delete existing Portlet preferences if not set in the Application Update Request.

```suggestion
    Portlet portletPreferences = new Portlet();
    preferences.forEach(preference -> portletPreferences.setValue(preference.getName(), preference.getValue()));
```

Reply:
Ok, I rollback this point. I don't remember why I change it.

Reply:
option keepExisting added, with default value=false

done